### PR TITLE
Adding source to the ChatMessage data model 

### PR DESCRIPTION
**Pull request by dvargas92495**
For use here: https://github.com/vellum-ai/vellum-example-apps/pull/4/files#diff-d6d2bfcd1d7464af75abb25aee78074204c39e317b7fb10e53b028d08b14a03eL155-R156

### CODE REVIEW

File: fern/openapi/openapi.yml [hunk 1]
```diff
@@ -1317,6 +1341,11 @@ components:
           allOf:
           - $ref: '#/components/schemas/ChatMessageContent'
           nullable: true
+        external_id:
+          type: string
+          nullable: true
+          description: An optional identifier representing the user or invocation
+            that generated the message.
```
Comment (noanflaherty) [inline fern/openapi/openapi.yml:1348]: IMO this description should include the scope of uniqueness. Could be as simple as "This id should be unique across all chat messages within a given chat history."

**Review by noanflaherty — APPROVED** [review]

File: fern/openapi/openapi.yml (hunk 1 above)
Comment (dvargas92495) [reply fern/openapi/openapi.yml:1348]: > This id should be unique across all chat messages within a given chat history.

This isn't true though. For example for user messages, it's used to say which User messages are me vs you, for example

Comment (dvargas92495) [reply fern/openapi/openapi.yml:1348]: Again, to clarify, it identifies the _source_ that generated the message. _Not_ the message itself

**Comment by dvargas92495** [comment]
After some back and forth offline, we decided to change the field back to `source` over `external_id`. Here's the full context:
Screenshot 2024-03-13 at 11 01 08 AM

**Pull request merged**